source/fe/fe_nedelec.cc

```diff
@@ -503,72 +492,60 @@ FE_Nedelec<dim>::initialize_quad_dof_index_permutation_and_sign_change()
   const unsigned int half_dofs = k * (k + 1); // see below;
 
   const int rl = row_length[k];
-
-  int offset = table_size[0];
-  // The assignment above is to prevent the compiler from complaining about the
-  // unused table_size.
-
-  int value = 0;
-
-  for (const bool face_orientation : {false, true})
-    for (const bool face_rotation : {false, true})
-      for (const bool face_flip : {false, true})
+  for (types::geometric_orientation combined_orientation = 0;
+       combined_orientation <
+       this->reference_cell().n_face_orientations(face_no);
+       ++combined_orientation)
```

Comment:
I imagine there are now many such places, but one of the things that's a bit annoying about this code is that we used to loop over a discrete set and now loop over integers *that happen to correspond to specific orientations*. I think it might be nice if we had a way to say something like
```
for (types::geometric_orientation combined_orientation : this->reference_cell().all_possible_face_orientations())
```
or, shorter,
```
for (types::geometric_orientation combined_orientation : this->reference_cell().face_orientations())
```
This makes it clear that the `combined_orientations` need not be a contiguous set of integers.